lib/common.js

```diff
@@ -0,0 +1,98 @@
+var firefox = require('node-firefox');
+
+var common = module.exports = {
+
+  dumpJSON: function(data, options) {
+    console.dir(data, {
+      depth: null,
+      colors: !options.parent.nocolors
+    });
+  },
+
+  handleOutputOptions: function(data, options, handler) {
+    if (options.parent.quiet) {
+      return;
+    } else if (options.parent.json) {
+      return common.dumpJSON(data, options);
+    } else {
+      return handler();
+    }
+  },
+
+  intList: function(val) {
+    return val.split(',').map(function(item) {
+      return parseInt(item);
+    });
+  },
+
+  selectPorts: function(ports) {
+    return firefox.findPorts().then(function(availablePorts) {
+      var selectedPorts;
+      if (!ports || ports.length === 0) {
+        // By default, install to all available ports.
+        selectedPorts = availablePorts;
+      } else {
+        // Filter available ports by --ports option.
+        selectedPorts = availablePorts.filter(function(port) {
+          return options.ports.indexOf(port.port) !== -1;
```

Comment:
Just tell me which though, otherwise this is all r+ and I have a branch that adds the build tools and fixes a few other things. I'll tweak it then push to master :+1: